servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Completable.java

```diff
@@ -1537,6 +1537,24 @@ public final Completable liftAsync(CompletableOperator operator) {
      * @return A new {@link Completable} that will use the passed {@link Executor} to invoke all methods on the
      * {@link Subscriber}.
      */
+    public final Completable publishOn(io.servicetalk.concurrent.Executor executor) {
+        return publishOn((Executor) executor);
```

Comment:
I expected the opposite change: non-deprecated method should be the one that does the action, deprecated method should downcast from `api.Executor` to `io.servicetalk.concurrent.Executor` and invoke this method.

Otherwise, there is a risk of `ClassCastException` if users pass `io.servicetalk.concurrent.Executor` that does not implement `io.servicetalk.concurrent.api.Executor`

Reply:
Nvm, this is for 0.41 branch, not for main

Reply:
Using a downcast somewhere seems like the simplest approach and I don't think it is a particular risk. The alternative is to start changing all of the references in `PublishAndSubscribeOn` classes and all of the classes they touch. I looked at this briefly but decide it touched too many things and it does seem unlikely that there is a `io.servicetalk.concurrent.Executor` that is not a `io.servicetalk.concurrent.api.Executor` in current use.

Reply:
Agreed, the risk is low